server/src/main/java/org/elasticsearch/cluster/metadata/SingleNodeShutdownMetadata.java

```diff
@@ -102,6 +111,7 @@ private SingleNodeShutdownMetadata(
             throw new IllegalArgumentException("shard allocation delay is only valid for RESTART-type shutdowns");
         }
         this.allocationDelay = allocationDelay;
+        this.targetNodeName = targetNodeName;
```

Comment:
We validate that there is an allocation delay only specified for the RESTART type, so we should also validate that the target note is only specified for a `REPLACE` type shutdown here I think